store/orm/orm.go

```diff
@@ -377,6 +377,54 @@ func (orm *ORM) UnscopedJobRunsWithStatus(statuses ...models.RunStatus) ([]model
 	return runs, err
 }
 
+// UnscopedJobRunsWithStatusCB passes all JobRuns to a callback, one by one,
+// including soft deleted ones.
+func (orm *ORM) UnscopedJobRunsWithStatusCB(cb func(*models.JobRun) error, statuses ...models.RunStatus) error {
+	runs := []models.JobRun{}
+	err := orm.DB.Unscoped().Where("status IN (?)", statuses).Order("status asc").Find(&runs).Error
```

Comment:
As an alternative approach to this, since we're already doing multiple queries, maybe we can get away w the classic pluck w preload:

```golang
var ids []string
orm.DB.Unscoped().Where("status IN = (?)", statuses).Pluck("ID", &ids).Error
orm.DB.Unscoped().Where(ids).preLoadRuns().Find(&runs)
```